python/tests/test_loops.py

```diff
@@ -144,7 +144,7 @@ def f(out, i):
 def test_for_each_no_push(modes):
     @for_each_operator()
     def f(out, i, ctx):
-        out[i] += i + 1
+        out[i] = i + 1
```

Comment:
This test is definitely thread safe as it was before. Let's not change it without a reason.

Reply:
It failed in CI (twice).

https://github.com/KatanaGraph/katana/runs/2018400129?check_suite_focus=true

Reply:
It actually is different because there is no push so there can be no race. However in the end it doesn't really matter. I just don't like changing tests without a good reason.

Reply:
I don't mind reverting `+=` for `=` but it seems weird to me to write `+=` when you don't actually mean to accumulate anything and the sibling test is definitely was not safe with `+=`.

Reply:
I think I probably wrote it the original way to create more complex code to "test more", but it's not at all clear that's better. I may also have been thinking about defeating the optimizer, but again, it's not clear that's needed here.